src/main/kotlin/no/digdir/fdk/parseservice/extract/dataset/Dataset.kt

```diff
@@ -0,0 +1,34 @@
+package no.digdir.fdk.parseservice.extract.dataset
+
+import no.digdir.fdk.model.dataset.Dataset
+import no.digdir.fdk.parseservice.extract.containsTriple
+import no.digdir.fdk.parseservice.extract.extractCatalogData
+import no.digdir.fdk.parseservice.extract.extractHarvestMetaData
+import no.digdir.fdk.parseservice.extract.fdkRecord
+import no.digdir.fdk.parseservice.extract.primaryTopicFromFdkRecord
+import no.digdir.fdk.parseservice.extract.singleObjectStatement
+import no.digdir.fdk.parseservice.namespace.FDK
+import org.apache.jena.rdf.model.Model
+import org.apache.jena.vocabulary.DCAT
+import org.apache.jena.vocabulary.DCTerms
+
+fun Model.extractDataset(): Dataset {
+    val builder = Dataset()
+
+    val record = fdkRecord(listOf(DCAT.Dataset))
+    val dataset = record.primaryTopicFromFdkRecord(listOf(DCAT.Dataset))
+    val harvestMetaData = record.extractHarvestMetaData()
+
+    builder.id = record.singleObjectStatement(DCTerms.identifier)!!.string
+    builder.uri = dataset.uri
+    builder.harvest = harvestMetaData
+    builder.catalog = dataset.extractCatalogData()
+
+    builder.isRelatedToTransportportal = containsTriple(dataset.uri, FDK.isRelatedToTransportportal.uri, true)
+    builder.isOpenData = containsTriple(dataset.uri, FDK.isOpenData.uri, true)
+    builder.isAuthoritative = containsTriple(dataset.uri, FDK.isAuthoritative.uri, true)
+
+    builder.addV11Values(dataset)
```

Comment:
Synes at denne pattern er for pragmatisk og krever også kodeendringer i eksisterende kode. Det krever også separate metoder for hver versjon. Synes det er bedre med Strategy metoden. På denne måten kan man lage migrerings logikk, enklere å teste og enklere å gjøre endringer.

Så hadde jeg også forsøkt å lage konfigurasjondrevet tilnærming isteden for hardkodet extract kode. På denne måten krever det ikke kodeendringer og kan man for eks definere versjoner i json.

// config/dcat-v1.1.json
{
  "version": "1.1",
  "properties": [
    {
      "propertyUri": "http://purl.org/dc/terms/title",
      "targetField": "title",
      "extractor": "localizedStrings"
    }
  ]
}

Reply:
noe ala dette du tenker? 7ef674ab3adb73ac936d06cfb824d3e0f07ec08b